crates/utils/re_uri/src/lib.rs

```diff
@@ -23,64 +25,45 @@ impl TimeRange {
 }
 
 impl std::fmt::Display for TimeRange {
+    /// Used for formatting time ranges in URLs
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         let Self { timeline, range } = self;
-        write!(f, "{}@", timeline.name())?;
-        match timeline.typ() {
-            re_log_types::TimeType::Sequence => {
-                let min = range.min.floor().as_i64();
-                let max = range.max.ceil().as_i64();
-                write!(f, "{min}..{max}")
-            }
-            re_log_types::TimeType::Time => {
-                let min = range.min.as_secs_f64();
-                let max = range.max.as_secs_f64();
-                write!(f, "{min:.4}s..{max:.4}s")
-            }
-        }
+
+        let min = IndexCell::new(timeline.typ(), range.min.floor());
+        let max = IndexCell::new(timeline.typ(), range.max.ceil());
+
+        let name = timeline.name();
+        write!(f, "{name}@{min}..{max}")
     }
 }
 
 impl TryFrom<&str> for TimeRange {
     type Error = Error;
 
     fn try_from(value: &str) -> Result<Self, Self::Error> {
-        let (timeline, range) = value.split_once('@').ok_or(Error::InvalidTimeRange)?;
-
-        let (min, max) = range.split_once("..").ok_or(Error::InvalidTimeRange)?;
-
-        // NOTE: We have some well-known timeline names, but we're explicitly not using them here
-        //       as this URL should really not be constructed by a user, ever.
-        let (min, typ) = match min.strip_suffix('s') {
-            Some(min) => (min, re_log_types::TimeType::Time),
-            None => (min, re_log_types::TimeType::Sequence),
-        };
-        let (max, typ) = match (max.strip_suffix('s'), typ) {
-            // if `max` had no `s` suffix, but `min` does, then it's a temporal timeline:
-            (Some(max), re_log_types::TimeType::Sequence) => (max, re_log_types::TimeType::Time),
-            // otherwise it's just whatever we already had, with the suffix stripped:
-            (Some(max), typ) => (max, typ),
-            (None, typ) => (max, typ),
-        };
+        let (timeline, range) = value
+            .split_once('@')
+            .ok_or_else(|| Error::InvalidTimeRange("Missing @".to_owned()))?;
+
+        let (min, max) = range
+            .split_once("..")
+            .ok_or_else(|| Error::InvalidTimeRange("Missing ..".to_owned()))?;
+
+        let min = min.parse::<IndexCell>().map_err(|err| {
+            Error::InvalidTimeRange(format!("Failed to parse time index '{min}': {err}"))
+        })?;
+        let max = max.parse::<IndexCell>().map_err(|err| {
+            Error::InvalidTimeRange(format!("Failed to parse time index '{max}': {err}"))
+        })?;
+
+        if min.typ() != max.typ() {
+            return Err(Error::InvalidTimeRange(
+                "min/max had differing types".to_owned(),
```

Comment:
would be good to message what types were detected in case user doesn't realize what went wrong